boards/mega-xplained/board.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin, Hinnerk van Bruinehsen
+ *               2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2016 Laurent Navet <laurent.navet@gmail.com>
+ *               2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_mega-xplained
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementation for the Mega Xplained
+ *
+ * @author      Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Laurent Navet <laurent.navet@gmail.com>
+ * @author      Matthew Blue <matthew.blu.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <avr/io.h>
+
+#include "board.h"
+#include "cpu.h"
+#include "uart_stdio.h"
+#include "periph/gpio.h"
+
+void SystemInit(void);
+static int uart_putchar(char c, FILE *stream);
+static int uart_getchar(FILE *stream);
+
+static FILE uart_stdout = FDEV_SETUP_STREAM(uart_putchar, NULL, _FDEV_SETUP_WRITE);
+static FILE uart_stdin = FDEV_SETUP_STREAM(NULL, uart_getchar, _FDEV_SETUP_READ);
+
+void board_init(void)
+{
+    /* initialize stdio via USART_1 */
+    SystemInit();
+
+    /* initialize the CPU */
+    cpu_init();
+
+    irq_enable();
+}
+
+/**
```

Comment:
some stuff in this file seems to be duplicate with ```common/arduino-atmega/board.c```, maybe re-use?

Reply:
This was also my initial impression, see [first comment](https://github.com/RIOT-OS/RIOT/pull/8829#pullrequestreview-106701875). But as @ZetaR60 is pointing out, this board is not an arduino board.
To move faster, I proposed to fix that in a follow-up PR. This PR is already 99% ready IMHO (just missing the LED macros and initialization).